ui/ducks/send/helpers.js

```diff
@@ -133,27 +96,16 @@ export async function estimateGasLimitForSend({
   }
 
   try {
-    // Call into the background process that will simulate transaction
-    // execution on the node and return an estimate of gasLimit
-    const estimatedGasLimit = await estimateGas(paramsForGasEstimate);
-
-    const estimateWithBuffer = addGasBuffer(
-      estimatedGasLimit,
-      blockGasLimit,
+    const { gas } = await estimateGasBuffered(
+      paramsForGasEstimate,
       bufferMultiplier,
     );
-    return addHexPrefix(estimateWithBuffer);
+    return gas;
   } catch (error) {
-    const simulationFailed =
-      error.message.includes('Transaction execution error.') ||
-      error.message.includes(
-        'gas required exceeds allowance or always failing transaction',
-      ) ||
-      (CHAIN_ID_TO_GAS_LIMIT_BUFFER_MAP[chainId] &&
-        error.message.includes('gas required exceeds allowance'));
-    if (simulationFailed) {
+    if (error.simulationFails) {
+      const blockGasLimit = await getBlockGasLimit(global.ethQuery);
```

Comment:
Why is we've replaced the previous block gas calculation with the actual one?

Is it for more accuracy? Do we not need to respect the value passed in the options if provided?